searx/engines/crates.py

```diff
@@ -0,0 +1,70 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Cargo search on crates.io"""
+
+from collections import OrderedDict
+from urllib.parse import urlencode
+
+from dateutil import parser
+
+about = {
+    "website": "https://crates.io/",
+    "wikidata_id": None,
+    "official_api_documentation": "https://crates.io/data-access",
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+categories = ["it", "packages", "cargo"]
```

Comment:
You can further use `!cargo` .. when we implement other cargo reposetories we will add this engines also to group `cargo` .. so with `!cargo` we are able to search through all cargo managers in once.